src/main/java/httpserver/ResponseBuilder.java

```diff
@@ -0,0 +1,8 @@
+package httpserver;
+
+import httpserver.httpmessages.HTTPResponse;
+
+public interface ResponseBuilder {
+
+    HTTPResponse buildResponse(String method, String path);
```

Comment:
What is the difference between this interface and and the `RequestHandler`?
You can put a `method` and `path` (and a `body`!) together to form a `HttpRequest` which is exactly what the `RequestHandler` gives you.

I would refactor this interface away.